pb/p2pd.proto

```diff
@@ -56,3 +63,42 @@ message StreamInfo {
   required bytes addr = 2;
   required string proto = 3;
 }
+
+message DHTRequest {
+  enum Type {
+    FIND_PEER                    = 0;
+    FIND_PEERS_CONNECTED_TO_PEER = 1;
+    FIND_PROVIDERS               = 2;
+    GET_CLOSEST_PEERS            = 3;
+    GET_PUBLIC_KEY               = 4;
+    GET_VALUE                    = 5;
+    SEARCH_VALUE                 = 6;
+    PUT_VALUE                    = 7;
+    PROVIDE                      = 8;
+  }
+
+  required Type type = 1;
+  optional bytes peer = 2;
+  optional bytes cid = 3;
+  optional string key = 4;
+  optional bytes value = 5;
+  optional int32 count = 6;
+  optional int64 timeout = 7;
+}
+
+message DHTResponse {
```

Comment:
I think we need a token that identifies the DHT request uniquely. Then all streaming responses would refer to that token, and the `END` response would close the life of the request. Currently it seems that responses for parallel DHT requests could get intermixed on the wire.

Reply:
I thought about it too and concluded that request multiplexing is more trouble than it's worth.
If we want parallel queries, we can open another connection and issue the query there.
unix sockets are relatively cheap, just a transient file descriptor.

Reply:
I guess that makes sense, although it complicates the client because it'll either need to sync to allow only 1 concurrent request (bad), or it'll need to maintain a pool of client instances (too complicated), because we don't want _each_ request to create a new client...

Reply:
We can create a new socket for each request, that's simple enough.